src/plugins/intel_gpu/src/graph/shape_of.cpp

```diff
@@ -24,7 +24,8 @@ layout shape_of_inst::calc_output_layout(shape_of_node const& node, kernel_impl_
         dt = impl_param.get_fused_output_layout().data_type;
     }
 
-    cldnn::tensor out_size{static_cast<tensor::value_type>(prim->output_rank), 1, 1, 1};
+    auto in_shape = impl_param.get_input_layout();
+    cldnn::tensor out_size{static_cast<tensor::value_type>(in_shape.get_rank()), 1, 1, 1};
```

Comment:
Why? With old shape infer input rank may be extended, thus produced output shape may be invalid with such change.

Reply:
With the definition of [ShapeOf](https://docs.openvino.ai/2022.3/openvino_docs_ops_shape_ShapeOf_3.html), the outpu number of elements is equal to input tensor rank, but it was output_rank, and it was 1 in this case. So it is fixed with input rank. But as you say that input rank may be extended, then should I use other approach?

Reply:
> then should I use other approach?

Yes, I think you should replace out rank with in rank here: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/plugin/ops/shape_of.cpp#L22
and maybe rename parameter from output_rank to input_rank in the primitive

Reply:
I tried to update as you suggested. And could you check it please? Thanks for your comments.